var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -108,10 +159,12 @@ class Trilinos(CMakePackage):
     depends_on('glm')
     depends_on('metis@5:', when='+metis')
     depends_on('suite-sparse', when='+suite-sparse')
+    depends_on('zlib', when="+zlib")
 
     # MPI related dependencies
     depends_on('mpi')
     depends_on('netcdf+mpi')
+    depends_on('parallel-netcdf', when="+pnetcdf@master")
```

Comment:
Assuming that this functionality appears in the next release, you need to change this line to:
```
depends_on('parallel-netcdf', when="+pnetcdf@master")
depends_on('parallel-netcdf', when="+pnetcdf@12.10.2:")
```
(note the non-existing `12.10.2` above). The second line would also work for `develop`.